appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/PayaraMicro.java

```diff
@@ -1410,6 +1429,34 @@ private void scanArgs(String[] args) {
                     case "--disablePhoneHome":
                         disablePhoneHome = true;
                         break;
+                    case "--enableRequestTracing":
+                        enableRequestTracing = true;
+                        break;
+                    case "--requestTracingThresholdUnit":
+                        try {
+                            TimeUnit.valueOf(args[i + 1].toUpperCase());
+                            requestTracingThresholdUnit = 
+                                    args[i + 1].toUpperCase();
+                        } catch (IllegalArgumentException e) {
+                            logger.log(Level.WARNING, "{0} is not a valid value"
+                                    + " for --requestTracingThresholdUnit", 
+                                    requestTracingThresholdUnit);
+                            throw e;
+                        }
+                        i++;
+                        break;
+                    case "--requestTracingThresholdValue":
+                        try {
+                            requestTracingThresholdValue = 
+                                    Long.parseLong(args[i + 1]);
```

Comment:
this smells like `ArrayIndexOutOfBoundsException` when a user forgets to include the actual units.